app/models/user.rb

```diff
@@ -1,11 +1,15 @@
-class User < ApplicationRecord
-  has_many :posts, foreign_key: 'author_id'
-  has_many :comments, foreign_key: 'author_id'
-  has_many :likes, foreign_key: 'author_id'
-  validates :name, presence: true
-  validates :posts_counter, numericality: { greater_than_or_equal_to: 0 }
-
-  def recent_posts
-    posts.order(created_at: :DESC).limit(3)
-  end
-end
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+  validates :name, presence: true
+  validates :posts_counter, numericality: { greater_than_or_equal_to: 0 }
```

Comment:
- [ ] still unable to login and sign up and because of line 10 when registering a user the request rollback because the posts_counter at that moment is nil and in the validation you say only numeric values 
you can modify the column `posts_counter` and make a default values as 0 and that will solve your problem you can find [here how to do it ](https://stackoverflow.com/questions/1550688/rails-how-do-i-create-a-default-value-for-attributes-in-rails-activerecords-mo)

Reply:
Thanks a lot. I was able to fix it now. :)